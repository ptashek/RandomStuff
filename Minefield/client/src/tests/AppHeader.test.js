import React from 'react';
import AppHeader from '../components/AppHeader.react';
import renderer from 'react-test-renderer';

it('renders ok with default props', () => {
  const tree = renderer.create(<AppHeader />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders ok with animate=false and no children', () => {
  const tree = renderer.create(
    <AppHeader animate={false} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders ok with animate=true and no children', () => {
  const tree = renderer.create(
    <AppHeader animate={true} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders ok with animate=true and one child', () => {
  const tree = renderer.create(
    <AppHeader animate={true}>
      <div>test</div>
    </AppHeader>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders ok with animate=false and one child', () => {
  const tree = renderer.create(
    <AppHeader animate={false}>
      <div>test</div>
    </AppHeader>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders ok with animate=true and many children', () => {
  const tree = renderer.create(
    <AppHeader animate={true}>
      <div>one</div>
      <div>two</div>
    </AppHeader>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders ok with animate=false and many children', () => {
  const tree = renderer.create(
    <AppHeader animate={false}>
      <div>one</div>
      <div>two</div>
    </AppHeader>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
