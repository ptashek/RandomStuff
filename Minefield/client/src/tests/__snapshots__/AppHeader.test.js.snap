exports[`test renders ok with animate=false and many children 1`] = `
<div>
  <img
    alt="logo"
    className="App-Header-Logo App-Logo"
    src="logo.svg" />
  <div
    className="App-Header-Text">
    <div>
      one
    </div>
    <div>
      two
    </div>
  </div>
</div>
`;

exports[`test renders ok with animate=false and no children 1`] = `
<div>
  <img
    alt="logo"
    className="App-Header-Logo App-Logo"
    src="logo.svg" />
  <div
    className="App-Header-Text" />
</div>
`;

exports[`test renders ok with animate=false and one child 1`] = `
<div>
  <img
    alt="logo"
    className="App-Header-Logo App-Logo"
    src="logo.svg" />
  <div
    className="App-Header-Text">
    <div>
      test
    </div>
  </div>
</div>
`;

exports[`test renders ok with animate=true and many children 1`] = `
<div>
  <img
    alt="logo"
    className="App-Header-Logo App-Logo App-Logo-Animated"
    src="logo.svg" />
  <div
    className="App-Header-Text">
    <div>
      one
    </div>
    <div>
      two
    </div>
  </div>
</div>
`;

exports[`test renders ok with animate=true and no children 1`] = `
<div>
  <img
    alt="logo"
    className="App-Header-Logo App-Logo App-Logo-Animated"
    src="logo.svg" />
  <div
    className="App-Header-Text" />
</div>
`;

exports[`test renders ok with animate=true and one child 1`] = `
<div>
  <img
    alt="logo"
    className="App-Header-Logo App-Logo App-Logo-Animated"
    src="logo.svg" />
  <div
    className="App-Header-Text">
    <div>
      test
    </div>
  </div>
</div>
`;

exports[`test renders ok with default props 1`] = `
<div>
  <img
    alt="logo"
    className="App-Header-Logo App-Logo"
    src="logo.svg" />
  <div
    className="App-Header-Text" />
</div>
`;
